tests/NotificationTests/NotificationCenterTest.php

```diff
@@ -0,0 +1,639 @@
+        $eBuilder = new EventBuilder;
+        $callbackInput = array($eBuilder, 'createImpressionEvent');
+
+        $this->loggerMock->expects($this->at(0))
+            ->method('log')
+            ->with(Logger::INFO, "Callback added for notification type '{$notificationType}'.");
+        $this->assertSame(
+            3,
+            $this->notificationCenterObj->addNotificationListener($notificationType, $callbackInput)
+        );
+        // verify that notifications length has incremented by 1
+        $this->assertSame(
+            3,
+            sizeof($this->notificationCenterObj->getNotifications()[$notificationType])
+        );
+    }
+
+    public function testAddNotificationForMultipleNotificationTypes()
+    {
+        $this->notificationCenterObj->clearAllNotifications();
+
+        // ===== should add, log and return notification ID when a valid callback is added for each notification type =====
+        $this->loggerMock->expects($this->at(0))
+            ->method('log')
+            ->with(Logger::INFO, sprintf("Callback added for notification type '%s'.", NotificationType::DECISION));
+        $this->assertSame(
+            1,
+            $this->notificationCenterObj->addNotificationListener(NotificationType::DECISION, function () {
+            })
+        );
+        // verify that notifications length for NotificationType::DECISION has incremented by 1
+        $this->assertSame(
+            1,
+            sizeof($this->notificationCenterObj->getNotifications()[NotificationType::DECISION])
+        );
+
+        $this->loggerMock->expects($this->at(0))
+            ->method('log')
+            ->with(Logger::INFO, sprintf("Callback added for notification type '%s'.", NotificationType::TRACK));
+        $this->assertSame(
+            2,
+            $this->notificationCenterObj->addNotificationListener(NotificationType::TRACK, function () {
+            })
+        );
+        // verify that notifications length for NotificationType::TRACK has incremented by 1
+        $this->assertSame(
+            1,
+            sizeof($this->notificationCenterObj->getNotifications()[NotificationType::TRACK])
+        );
+
+        $this->loggerMock->expects($this->at(0))
+            ->method('log')
+            ->with(Logger::INFO, sprintf("Callback added for notification type '%s'.", NotificationType::FEATURE_ACCESSED));
+        $this->assertSame(
+            3,
+            $this->notificationCenterObj->addNotificationListener(NotificationType::FEATURE_ACCESSED, function () {
+            })
+        );
+        // verify that notifications length for NotificationType::FEATURE_ACCESSED has incremented by 1
+        $this->assertSame(
+            1,
+            sizeof($this->notificationCenterObj->getNotifications()[NotificationType::FEATURE_ACCESSED])
+        );
+    }
+
+    public function testAddNotificationForMultipleCallbacksForANotificationType()
```

Comment:
just pointing out here, will chk later. We have to check if same method added twice, it should return -1.